File: RPG-Git/src/pokemon/database/Query.java
package pokemon.database;

import java.util.Collections;
import java.util.HashMap;
import java.util.Map;
import pokemon.Category;
import pokemon.Type;

/**
 *
 * @author evinstk
 */
public class Query {
    public static String getName(int index) {
        return "Bulbasaur";
    }
    
    public static Map<String, Short> getBaseStats(int index) {
        Map<String, Short> builder = new HashMap<String, Short>(5);
        
        //Dummy code. These are Bulbasaur's base stats.
        builder.put("HP", (short)45);
        builder.put("ATTACK", (short)49);
        builder.put("DEFENSE", (short)49);
        builder.put("SPEED", (short)45);
        builder.put("SPECIAL", (short)65);
        Map<String, Short> ret = Collections.unmodifiableMap(builder);
        return ret;
    }
    
    public static Type[] getType(int index) {
        //Dummy code
        Type [] ret = {Type.GRASS, Type.NO_TYPE};
        return ret;
    }
    
    public static String getExperienceGroup(String name) {
        return "MediumSlow";
    }
    
    public static Object[] getMoveData(int index) {
        Object[] ret = {1, "Pound", Type.NORMAL, Category.PHYSICAL, 35, 40, 100, false};
        return ret;
    }
}

File: PokemonRedBlue/query.cpp
/*
 * File:   query.cpp
 * Author: evinstk
 * 
 * Created on June 17, 2013, 9:15 AM
 */

#include <string>
#include "query.h"
#include "move.h"
#include "type.h"
#include "category.h"

string *query::getName(unsigned char index) {
    string *ret = new string("Bulbasaur");
    return ret;
}

unsigned char *query::getBaseStats(unsigned char index) {
    unsigned char *ret = new unsigned char[5];
    ret[0] = 45;
    ret[1] = 49;
    ret[2] = 49;
    ret[3] = 45;
    ret[4] = 65;
    return ret;
}

char query::getExpGroup(unsigned char index) {
    return 2;
}

vector<Move *> query::getDefaultMoves(unsigned char index) {
    vector<Move *> ret;
    Move *move = new Move(1);
    ret.push_back(move);
    return ret;
}

Type *query::getTypes(unsigned char index) {
    Type *ret = new Type[2];
    ret[0] = GRASS;
    ret[1] = NO_TYPE;
    return ret;
}

string *query::getMoveName(unsigned char move_index) {
    string *ret = new string("Pound");
    return ret;
}

Type query::getType(unsigned char move_index) {
    Type ret = NORMAL;
}

Category query::getCategory(unsigned char move_index) {
    return PHYSICAL;
}

unsigned char query::getMaxPp(unsigned char move_index) {
    return 35;
}

unsigned char query::getPower(unsigned char move_index) {
    return 40;
}

unsigned char query::getAccuracy(unsigned char move_index) {
    return 100;
}

bool query::getCrRatio(unsigned char move_index) {
    return false;
}